[Elijah5399] iP

Elijah5399 opened the pull request:
# Alyssa

> "I'm going to become king of the pirates" - Monkey D Luffy

Alyssa frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~fast~~ _SUPER_ FAST to use

All you need to do is:

1. download it from [here](https://github.com/Elijah5399/ip/releases/download/A-Jar/MYAlyssa.jar)
2. double click it
3. add your tasks
4. let it manage your tasks for you! :+1:

And it is **FREE**!
Features:

- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders (coming soon)

---

If you are a Java programmer, you can use it to practice your Java too. Here's the `main` method:

```java
public static void main(String[] args) {
    new Alyssa("./data/alyssa.txt", "./data").execute();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Elijah5399 committed:
Added new logo and initial messages

Elijah5399 committed:
Level 1

Elijah5399 committed:
Level-2 completed

Elijah5399 committed:
Level-3

Elijah5399 committed:
Added todos, events and deadlines

Elijah5399 committed:
Added automated text UI testing

Elijah5399 committed:
Added error handling

Elijah5399 committed:
Completed level 6

Elijah5399 committed:
Added enums and organised main function

Elijah5399 committed:
Organised code

Elijah5399 committed:
Added custom exceptions

Elijah5399 committed:
Add hard disk saving and loading functionality to Alyssa

Elijah5399 committed:
Fix bugs in saving and loading file

Elijah5399 committed:
Merge branch 'branch-Level-7'

Elijah5399 committed:
Implemented Date recognition for deadlines

Elijah5399 committed:
Merge branch 'branch-Level-8'

Elijah5399 committed:
Organise code into various files

Elijah5399 committed:
Make source code part of the alyssa package

Elijah5399 committed:
Add /alyssa inside /java

Elijah5399 committed:
Merge branch 'add-gradle-support'

Elijah5399 committed:
Add tests for methods in TaskList and Deadline

Elijah5399 committed:
Modify build.gradle to produce jar files correctly

Elijah5399 committed:
Add a few JavaDoc

Elijah5399 committed:
Implement level 9 - finding of tasks

Elijah5399 committed:
Merge with branch level 9

Elijah5399 committed:
Change constant variable names

Elijah5399 committed:
Merge branch 'branch-A-Javadoc'

Elijah5399 committed:
Add GUI support

yytan25 commented on src/main/java/alyssa/Deadline.java:
```diff
@@ -0,0 +1,38 @@
+ * A Deadline is a task with an end date/time.
+ */
+public class Deadline extends Task {
+    protected LocalDate by;
```
It's not immediately clear that your parameter `by` represents a date. Perhaps a name such as `byDate` or `dueDate` would be more clear to readers? (This also applies to your `LocalDate` parameters in `Event`)

yytan25 commented on src/main/java/alyssa/Ui.java:
```diff
@@ -0,0 +1,43 @@
+    /**
+     * Returns a goodbye message.
+     */
+    protected String goodbye() {
```
I understand that the intention is obvious, but since this is a method, perhaps it's more suitable to name it as `sayGoodbye` or `printGoodbye`?

yytan25 reviewed (commented):
LGTM except for a couple of nits. I really appreciate your attention to detail in using Javadocs for all your methods, and the additional comments help readers better understand your code. Keep up the good work!

SynapseProgramming commented on src/main/java/alyssa/Ui.java:
```diff
@@ -4,33 +4,33 @@
 public class Ui {
-    private static final String line = "____________________________________________________________";
-    private static final String logo = " ___  __    __   __  ____   ____    ___\n"
+    private static final String LINE = "____________________________________________________________";
+    private static final String LOGO = " ___  __    __   __  ____   ____    ___\n"
```
Good job, for changing static variables to be of full caps

SynapseProgramming replied:
Good job, for changing static variables to be of full caps.

SynapseProgramming commented on src/main/java/alyssa/Command.java:
```diff
@@ -5,28 +5,32 @@
+        case "delete":
+            return Command.DELETE;
+        default:
+            return Command.INVALID;
```
Good Job, for adhering to the indentation requirements for switch structures.

SynapseProgramming commented on src/main/java/Alyssa.java:
Maybe in the future, this commit could be broken down into many smaller commits, as it is really hard to keep track of so many changes in 1 commit.

Elijah5399 committed:
Remove unnecessary lines and functions

Elijah5399 committed:
Merge branch 'branch-level-10'

Elijah5399 committed:
Refactor code in Alyssa.java

The current run() method in Alyssa.java is very long
due to each switch case being multiple lines.

Extracting the functionality in each case allows
our run() function to be less condensed,
allowing our code to be more readable.

Elijah5399 committed:
Add assertions

The current codebase lacks assertions to check for
invariants.

Integrating assertions checks that our program is
working as expected, and catching exceptions
correctly.

Elijah5399 committed:
Merge pull request #1 from Elijah5399/branch-A-CodeQuality

Refactor code in Alyssa.java

Elijah5399 committed:
Merge branch 'master' into branch-A-Assertions

Elijah5399 committed:
Merge pull request #2 from Elijah5399/branch-A-Assertions

Add assertions

Elijah5399 committed:
Add help message

Elijah5399 committed:
Merge branch 'branch-C-Help'

Elijah5399 committed:
Update user guide

Elijah5399 committed:
Add title to GUI window

Elijah5399 committed:
Refactor goodbye() to sayGoodbye()

Elijah5399 committed:
Change Ui.png

Elijah5399 committed:
Remove dead code

Elijah5399 committed:
Change Ui.png to include more details

Elijah5399 committed:
Add italics in readme.md
